src/components/VizbeeMiniCastBar/hooks/useMiniCastBarMediaStatusListenerHook.js

```diff
@@ -0,0 +1,138 @@
+ * @param {string} props.buttonImgPlay - Image for the play button.
+ * @param {string} props.buttonImgPause - Image for the pause button.
+ * @param {string} props.buttonImgStop - Image for the stop button.
+ * @returns {Object} Object containing state variables and functions to handle media status updates.
+ */
+export const useMiniCastBarMediaStatusListenerHook = ({
+  isVisible,
+  setIsVisible,
+  playActionButtonType,
+  shouldDisablePlayPauseButtonForLive,
+  buttonImgPlay,
+  buttonImgPause,
+  buttonImgStop,
+}) => {
+  // Define state
+  const [mediaStatus, setMediaStatus] = useState({});
+  const [isButtonDisabled, setButtonDisabled] = useState(true);
+  const [buttonImg, setButtonImg] = useState(null);
+  const [buttonType, setButtonType] = useState(null);
+
+  // useEffect hook
+  useEffect(() => {
+    console.debug(`${LOG_TAG} Add VZB_MEDIA_STATUS Listener`);
+    const mediaStatusListener = VizbeeManager.addListener(
+      "VZB_MEDIA_STATUS",
+      videoPlayerStatusCallback
+    );
+
+    return () => {
+      console.debug(`${LOG_TAG} Remove VZB_MEDIA_STATUS Listener`);
+      VizbeeManager.removeAllListeners(mediaStatusListener);
+    };
+  }, []);
+
+  /**
+   * Update the state of the mini cast bar controls
+   *
+   * @param {Object} newMediaStatus - Media status containing video info
+   */
+  const updatePlayerState = (newMediaStatus) => {
+    console.debug(`${LOG_TAG} New Media Status = ${newMediaStatus}`);
+    if (newMediaStatus?.playerState === VizbeePlayerState.Playing) {
+      setButtonDisabled(false);
+
+      if (newMediaStatus?.isLive && playActionButtonType !== "playpause") {
+        setButtonType(ButtonType.STOP);
+        setButtonImg(buttonImgStop);
+      } else {
+        if (newMediaStatus?.isLive && shouldDisablePlayPauseButtonForLive) {
+          setButtonDisabled(true);
+        }
+
+        setButtonType(ButtonType.PAUSE);
+        setButtonImg(buttonImgPause);
+      }
+      return;
+    }
+
+    if (
+      newMediaStatus?.playerState === VizbeePlayerState.Paused ||
+      newMediaStatus?.playerState === VizbeePlayerState.Loading ||
+      newMediaStatus?.playerState === VizbeePlayerState.Buffering
+    ) {
+      if (newMediaStatus.isLive && shouldDisablePlayPauseButtonForLive) {
+        setButtonDisabled(true);
+      } else {
+        setButtonDisabled(false);
+      }
+
+      setButtonType(ButtonType.PLAY);
+      setButtonImg(buttonImgPlay);
+    }
+
+    // update the button state to disabled
+    if (mediaStatus.isAdPlaying) {
+      setButtonDisabled(true);
+    }
+  };
+
+  /**
+   * Callback for the VZB_MEDIA_STATUS Listener
+   * @param {Object} newMediaStatus - Media status containing video info
+   */
+  const videoPlayerStatusCallback = (newMediaStatus) => {
+    console.debug(
+      `${LOG_TAG} Media status current state :: ${newMediaStatus.playerState}`
+    );
+    if (newMediaStatus?.guid && newMediaStatus?.guid?.length != 0) {
```

Comment:
Same here: Change `if (newMediaStatus?.guid && newMediaStatus?.guid?.length != 0)` to `if(newMediaStatus?.guid?.length != 0)`